core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -113,6 +116,21 @@ public JoinDefinition define(AutoJoinerContext context) {
       } else {
         joinBuilder.setOutputSchemaName("join.output");
       }
+
+      if (conf.isDistributionEnabled()) {
```

Comment:
there's a lot of nesting here which makes it hard to read. Can you pull this out into a separate method?

Reply:
done